src/app/pages/TokenDashboardPage/TokenSupplyCard.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import { SnapshotCard } from '../../components/Snapshots/SnapshotCard'
+import { COLORS } from '../../../styles/theme/colors'
+import { useRequiredScopeParam } from '../../hooks/useScopeParam'
+import { useTokenInfo } from './hook'
+import { useLoaderData } from 'react-router-dom'
+
+export const TokenSupplyCard: FC = () => {
+  const { t } = useTranslation()
+  const scope = useRequiredScopeParam()
+
+  const address = useLoaderData() as string
+
+  const { token, isFetched } = useTokenInfo(scope, address)
+
+  const title = t('tokens.totalSupply')
+  const supplyString = token?.total_supply || '0'
+  const supplyNumber = parseInt(supplyString)
+
+  return (
+    <SnapshotCard title={title} label={token?.symbol}>
+      <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'center', height: '100%' }}>
+        {isFetched && (
+          <>
+            <Typography
+              component="span"
+              sx={{
+                fontSize: '48px',
+                fontWeight: 700,
+                color: COLORS.brandDark,
+              }}
+            >
+              {isNaN(supplyNumber)
```

Comment:
I don't expect it can be malformed, so I think `!supplyString` would be better

Reply:
refactored.